R/download_link.R

```diff
@@ -0,0 +1,130 @@
+#' Download link
+#'
+#' @param outputId The name of the output slot that the `downloadHandler` is assigned
+#' to.
+#' @param link_text Text that will appear describing the download action
+#' (default: "Download file").
+#' Vague text like 'click here' or 'here' will cause an error, as will ending in
+#' a full stop. Leading and trailing white space will be automatically trimmed.
+#' If the string is shorter than 7 characters a console warning will be thrown.
+#' There is no way to hush this other than providing more detail.
+#' @param file_type The file type to be download (default: CSV)
+#' @param file_size The file size if known. Needs to be a string ending in one
+#' of KB, MB, GB or rows.
+#'
+#' @returns shiny.tag object
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' ui <- fluidPage(
+#'   p("Choose a dataset to download."),
+#'   selectInput("dataset", "Dataset", choices = c("mtcars", "airquality")),
+#'   download_link("download_data", "Download data")
+#' )
+#'
+#' server <- function(input, output) {
+#'   # The requested dataset
+#'   data <- reactive({
+#'     get(input$dataset)
+#'   })
+#'
+#'   output$download_data <- downloadHandler(
+#'     filename = function() {
+#'       # Use the selected dataset as the suggested file name
+#'       paste0(input$dataset, ".csv")
+#'     },
+#'     content = function(file) {
+#'       # Write the dataset to the `file` that will be downloaded
+#'       write.csv(data(), file)
+#'     }
+#'   )
+#' }
+#'
+#' shinyApp(ui, server)
+#' }
+download_link <- function(
+    outputId,
+    link_text = "Download data",
```

Comment:
Does this not fail our own bad links checks? (if not, should it? It doesn't feel particularly descriptive)

Reply:
It's a fair cop. I've added `download data` to the banned list now and removed any default as well. If we're wanting to avoid generic text, I guess it's better not to give users an easy way out with a default value.